crates/control/src/obstacle_filter.rs

```diff
@@ -1,25 +1,25 @@
+use std::time::{Duration, SystemTime};
+
 use color_eyre::Result;
 use context_attribute::context;
+use filtering::KalmanFilter;
 use framework::{AdditionalOutput, HistoricInput, MainOutput, PerceptionInput};
-use nalgebra::{Isometry2, Point2};
+use itertools::{chain, iproduct};
+use nalgebra::{distance, point, Isometry2, Matrix2, Point2};
 use types::{
-    DetectedRobots, FieldDimensions, Obstacle, ObstacleFilterHypothesis, SensorData, SonarObstacle,
+    configuration::ObstacleFilter as ObstacleFilterConfiguration, CycleInfo, DetectedRobots,
+    FieldDimensions, Obstacle, ObstacleFilterHypothesis, ObstacleKind, SensorData, SonarObstacle,
 };
 
-pub struct ObstacleFilter {}
+pub struct ObstacleFilter {
+    hypotheses: Vec<ObstacleFilterHypothesis>,
+}
 
 #[context]
 pub struct CreationContext {
     pub field_dimensions: Parameter<FieldDimensions, "field_dimensions">,
-    pub goal_post_obstacle_radius:
-        Parameter<f32, "control.obstacle_filter.goal_post_obstacle_radius">,
-    // pub obstacle_filter:
-    //     Parameter<ObstacleFilter, "control.obstacle_filter">,
-    pub robot_obstacle_radius_at_foot_height:
-        Parameter<f32, "control.obstacle_filter.robot_obstacle_radius_at_foot_height">,
-    pub robot_obstacle_radius_at_hip_height:
-        Parameter<f32, "control.obstacle_filter.robot_obstacle_radius_at_hip_height">,
-    pub unknown_obstacle_radius: Parameter<f32, "control.obstacle_filter.unknown_obstacle_radius">,
```

Comment:
Why are these not needed anymore?

Reply:
They are needed, I just reordered them (hopefully)

Reply:
Ah, yes, they are not needed during creation. 👍